loader/0002-quilt.mod.json.md

```diff
@@ -0,0 +1,362 @@
+# Summary
+This document acts as a Request for Comments (RFC) on the topic of how a Quilt mod's metadata should be formatted. Such changes are intended to provide added value for mod developers, library developers, and users alike.
+
+# Motivation
+Since the toolchain is being forked, now is the opportune moment to make large scale changes to many of the systems mod developers use. This RFC describes a more flexible and powerful format for the mod file that defines what a mods definition should look like.
+
+# Explanation
+Below is an outline of all defined keys and values.
+
+* [schema_version](#the-schema_version-field) — The schemaVersion to be used for reading this file
+* [quilt_loader](#the-quilt_loader-field) — Information related to loading the mod
+    * [group_id](#the-group_id-field) — The Maven group id
+    * [mod_id](#the-mod_id-field) — The mod id
+    * [provides](#the-provides-field) — Alternative mod ids provided by this mod
+    * [version](#the-version-field) — The mods version
+    * [entrypoints](#the-entrypoints-field) — Collection of entrypoints
+    * [plugins](#the-plugins-field) — Collection of plugins
+    * [jars](#the-jars-field) — Array of nested JARs to be loaded
+    * [language_adapters](#the-language_adapters-field) — Array of language adapters
+    * [depends](#the-depends-field) — Collection of mod dependencies
+    * [breaks](#the-breaks-field) — Collection of mods that this mod is incompatible with
+    * [repositories](#the-repositories-field) — Array of maven repositories
+    * [metadata](#the-metadata-field) — Extra information about this mod and/or its authors
```

Comment:
How is the metadata related to loading? I think it should be moved outside the `quilt_loader` object.

Reply:
The way I see it the metadata for a mod specifically regards information needed to load the mod for quilt loader.

If you were to take the api surface route, quilt loader provides the apis to introspect this data so it makes sense here also.

Reply:
> The way I see it the metadata for a mod specifically regards information needed to load the mod for quilt loader.

Can't see the way it's associated.

> If you were to take the api surface route, quilt loader provides the apis to introspect this data so it makes sense here also.

The loader provides an API for any field in the `quilt.mod.json` file.

Reply:
Each top-level element in the mod.json field should belong to a specific mod, loader plugin, or game provider. `mixin` and `access_widener` would theoretically be loader plugins, while `minecraft` is a game provider. A top-level `metadata` field would mean there would be a mod/plugin/game named `metadata`.

To summarize, top-level fields do not indicate purpose, but rather ownership.